src/bp/ui-studio/src/web/actions/index.ts

```diff
@@ -480,3 +480,11 @@ export const getQnaCountByTopic = () => dispatch => {
     dispatch(receiveQNACountByTopic(data))
   })
 }
+
+export const receiveModuleTranslations = createAction('LANG/TRANSLATIONS')
+export const getModuleTranslations = () => dispatch => {
+  // tslint:disable-next-line: no-floating-promises
+  axios.get(`${window.API_PATH}/modules/translations`).then(({ data }) => {
+    dispatch(receiveModuleTranslations(data))
+  })
+}
```

Comment:
Here we request the translations to the backend, to then dispatch something that provokes and action that modifies a state that provokes and updates that adds these translations using `lang.extend`. I'm a bit confused why we wouldn't want to just write `lang.extend()` after we received the data?

Reply:
Because there's some concurrent issue when importing ui-shared directly in this file (a Blueprint dependency issue), and I spent some time to try to fix this without result, so I let it on the side for the moment